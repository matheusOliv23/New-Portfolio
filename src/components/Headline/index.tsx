import Button from "../Button"
import {
  ButtonContainerStyls,
  HeadlineContainer,
  SubtitleStyles,
  TextStyles,
  TitleStyles
} from "./styles"

export default function Headline() {

  function scrollDown() {
    window.scrollTo({ top: 800, behavior: "smooth" })
  }
  return (
    <div className={HeadlineContainer}>
      <h1 className={TitleStyles}>MATHEUS HENRIQUE OLIVEIRA</h1>
      <h2 className={SubtitleStyles}>
        Desenvolvedor <strong className="text-primary">Front-End</strong>
      </h2>
      <h3 className={TextStyles}>
        Transformando sonhos em realidade por meio da tecnologia
      </h3>
      <div className={ButtonContainerStyls}>
        <Button onClick={scrollDown} type="button" variant="PRIMARY">
          Quem sou eu
        </Button>
      </div>
    </div>
  )
}
