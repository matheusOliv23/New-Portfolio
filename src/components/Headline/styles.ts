export const HeadlineContainer = `flex flex-col lg:items-start z-50 gap-8 !opacity-100 items-center justify-center lg:text-start text-center`

export const TitleStyles = `font-bold md:text-5xl text-2xl tracking-widest`

export const SubtitleStyles = `lg:text-5xl text-4xl text-[#8a8f98] font-medium tracking-widest mt-2`

export const TextStyles = `lg:text-3xl text-2xl max-w-[32rem]`

export const ButtonContainerStyls = `flex gap-6 justify-center items-center`
