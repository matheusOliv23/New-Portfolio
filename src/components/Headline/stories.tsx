import { ComponentStory, ComponentMeta } from "@storybook/react"

import Headline from "./index"

export default {
  title: "Headline",
  component: Headline
} as ComponentMeta<typeof Headline>

const Template: ComponentStory<typeof Headline> = () => <Headline />

export const Default = Template.bind({})
Default.args = {
  title: "Headline"
}
