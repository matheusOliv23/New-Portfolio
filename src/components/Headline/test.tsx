import { render, screen } from "@testing-library/react"

import Headline from "./index"

const { getByRole, getByText } = screen

describe("<Headline />", () => {
  it("should render the Headline", () => {
    render(<Headline />)

    const title = getByRole("heading", { name: "MATHEUS HENRIQUE OLIVEIRA" })
    const subtitle = getByRole("heading", { name: "Desenvolvedor Front-End" })
    const text = getByText(
      "Transformando sonhos em realidade por meio da tecnologia"
    )

    expect(title).toBeInTheDocument()
    expect(subtitle).toBeInTheDocument()
    expect(text).toBeInTheDocument()
  })

  it("should match snapshot", () => {
    const { container } = render(<Headline />)

    expect(container).toMatchSnapshot()
  })
})
