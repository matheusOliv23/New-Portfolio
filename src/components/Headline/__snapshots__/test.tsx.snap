// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Headline /> should match snapshot 1`] = `
<div>
  <div
    class="flex flex-col lg:items-start z-50 gap-8 !opacity-100 items-center justify-center lg:text-start text-center"
  >
    <h1
      class="font-bold md:text-5xl text-2xl tracking-widest"
    >
      MATHEUS HENRIQUE OLIVEIRA
    </h1>
    <h2
      class="lg:text-5xl text-4xl text-[#8a8f98] font-medium tracking-widest mt-2"
    >
      Desenvolvedor 
      <strong
        class="text-primary"
      >
        Front-End
      </strong>
    </h2>
    <h3
      class="lg:text-3xl text-2xl max-w-[32rem]"
    >
      Transformando sonhos em realidade por meio da tecnologia
    </h3>
    <div
      class="flex gap-6 justify-center items-center"
    >
      <button
        class="ease-in duration-200 tracking-[0.25rem] before:bg-[#27282c] uppercase ring-2 ring-inset hover:text-primary relative text-2xl font-bold duration-300  hover:shadow-2xl hover:text-white  py-4 px-7 rounded min-w-[3rem]  text-center flex items-center w-full justify-center bg-[#1e9bff] text-white undefined"
        type="button"
      >
        Quem sou eu
      </button>
    </div>
  </div>
</div>
`;
